pancancer_utilities/tcga_utilities.py

```diff
@@ -0,0 +1,180 @@
+"""
+Functions for preprocessing TCGA expression data and mutation status labels.
+
+Most functions are adapted from:
+https://github.com/greenelab/BioBombe/blob/master/9.tcga-classify/scripts/tcga_util.py
+"""
+import os
+
+import pandas as pd
+from sklearn.preprocessing import StandardScaler
+
+def process_y_matrix(
+    y_mutation,
+    y_copy,
+    include_copy,
+    gene,
+    sample_freeze,
+    mutation_burden,
+    filter_count,
+    filter_prop,
+    output_directory,
+    hyper_filter=5,
+):
+    """
+    Combine copy number and mutation data and filter cancer-types to build y matrix
+
+    Arguments
+    ---------
+    y_mutation: Pandas DataFrame of mutation status
+    y_copy: Pandas DataFrame of copy number status
+    include_copy: boolean if the copy number data should be included in status calc
+    gene: string indicating gene of interest (used for writing proportion file)
+    sample_freeze: pandas dataframe storing which samples to use
+    mutation_burden: pandas dataframe storing log10 mutation counts
+    filter_count: the number of positives or negatives required per cancer-type
+    filter_prop: the proportion of positives or negatives required per cancer-type
+    output_directory: the name of the directory to store the gene summary
+    hyper_filter: the number of std dev above log10 mutation burden to filter
+
+    Returns
+    -------
+    Write file of cancer-type filtering to disk and output a processed y vector
+    """
+    if include_copy:
+        y_df = y_copy + y_mutation
+    else:
+        y_df = y_mutation
+
+    y_df.loc[y_df > 1] = 1
```

Comment:
I'm not very familiar with this type of data, so maybe this is obvious. But why are you setting a cap to 1?

Reply:
At this point, `y_df` contains information about how many deleterious mutations or copy number alterations were observed in the given gene for the given sample. We want to make this into a classification problem (either a sample has or doesn't have a mutation), so anything with 1 or more mutations gets a label of 1.

It would be interesting to treat this instead as an ordinal regression problem, i.e. predicting the number of mutations/CNVs from gene expression, but this might get a bit complicated since there are lots of subtle biases that go into the number of mutations (background mutation rates vary a lot across cancers and across samples, etc) and interpreting the results would be harder. Interesting future question, though!